### PR TITLE
[Doc] update limits for auto-increment in 3.0rc02 and merge comments from pm

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: docs/sql-reference/sql-statements/auto_increment.md [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
+Since version 3.0, StarRocks supports the `AUTO_INCREMENT` column attribute, which can simplify data management. This topic introduces the application scenarios, usage and features of the `AUTO_INCREMENT` column attribute.
 
 ## Introduction
 
-When a new data record is inserted, StarRocks automatically assigns a globally unique integer value for the record's `AUTO_INCREMENT` column as its unique ID, and the subsequent values for the `AUTO_INCREMENT` column automatically increase at a specific step starting from the ID of the record. `AUTO_INCREMENT` columns can be used to simplify data management and speed up some queries. Here are some application scenarios of `AUTO_INCREMENT` columns:
+When a new data row is loaded into a table, StarRocks automatically assigns a globally unique integer value for the row's `AUTO_INCREMENT` column as its unique ID within the table, and the subsequent values for the `AUTO_INCREMENT` column automatically increase at a specific step starting from the ID of the row. An `AUTO_INCREMENT` colum can be used to simplify data management and speed up some queries. Here are some application scenarios of an `AUTO_INCREMENT` column:
```
Comment (jaogoy) [inline docs/sql-reference/sql-statements/auto_increment.md:7]: 有个单词写成 colum

File: docs/sql-reference/sql-statements/auto_increment.md [hunk 2]
```diff
@@ -1,32 +1,32 @@
...
+Since version 3.0, StarRocks supports the `AUTO_INCREMENT` column attribute, which can simplify data management. This topic introduces the application scenarios, usage and features of the `AUTO_INCREMENT` column attribute.
 
 ## Introduction
 
-When a new data record is inserted, StarRocks automatically assigns a globally unique integer value for the record's `AUTO_INCREMENT` column as its unique ID, and the subsequent values for the `AUTO_INCREMENT` column automatically increase at a specific step starting from the ID of the record. `AUTO_INCREMENT` columns can be used to simplify data management and speed up some queries. Here are some application scenarios of `AUTO_INCREMENT` columns:
+When a new data row is loaded into a table, StarRocks automatically assigns a globally unique integer value for the row's `AUTO_INCREMENT` column as its unique ID within the table, and the subsequent values for the `AUTO_INCREMENT` column automatically increase at a specific step starting from the ID of the row. An `AUTO_INCREMENT` column can be used to simplify data management and speed up some queries. Here are some application scenarios of an `AUTO_INCREMENT` column:
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/auto_increment.md:7]: image
这个 within 感觉用 across 会更合适一些

File: docs/sql-reference/sql-statements/auto_increment.md [hunk 3]
```diff
@@ -1,32 +1,32 @@
...
 
-- Serve as primary keys: `AUTO_INCREMENT` columns can be used as primary keys to ensure that each record has a unique ID and make it easy to query and manage data.
-- Join tables: When multiple tables are joined, `AUTO_INCREMENT` columns can be used as Join Keys, which can expedite queries compared to using columns whose data types are UUID and STRING.
-- Count the number of distinct values in a high-cardinality column: An `AUTO_INCREMENT` column can be used to represent the unique value column in a dictionary. Compared to directly counting distinct STRING values, counting distinct integer values of the`AUTO_INCREMENT` column can sometimes improve the query speed by several times or even tens of times.
+- Serve as primary keys: An `AUTO_INCREMENT` column can be used as the primary key to ensure that each row has a unique ID and make it easy to query and manage data.
+- Join tables: When multiple tables are joined, an `AUTO_INCREMENT` column can be used as the Join Key, which can expedite queries compared to using a column whose data type is STRING, for example UUID.
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/auto_increment.md:10]: image
for example 后面要加逗号

File: docs/sql-reference/sql-statements/auto_increment.md [hunk 4]
```diff
@@ -95,85 +98,85 @@ SELECT * FROM t;
 
 ### Uniqueness
 
-In general, StarRocks guarantees that the values for the `AUTO_INCREMENT` columns are globally unique. We recommend that you do not explicitly assign and implicitly specify the values for the `AUTO_INCREMENT` columns at the same time. If you do so, it may break the global uniqueness of auto-incremented IDs. Here is a simple example: Create a table named `t` with two columns, `id` and `number`. Specify the column `number` as the `AUTO_INCREMENT` column.
+In general, StarRocks guarantees that the values for a `AUTO_INCREMENT` column are globally unique within a table. We recommend that you do not explicitly assign and implicitly specify the values for the `AUTO_INCREMENT` column at the same time. If you do so, it may break the global uniqueness of auto-incremented IDs. Here is a simple example: Create a table named `test_tbl2` with two columns, `id` and `number`. Specify the column `number` as the `AUTO_INCREMENT` column.
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/auto_increment.md:101]: image
同上。改的话这里同步

File: docs/sql-reference/sql-statements/auto_increment.md [hunk 5]
```diff
@@ -253,13 +256,13 @@ If the `AUTO_INCREMENT` column is not a primary key or a part of the primary key
 - If the row already exists in the table, StarRocks does not update the auto-incremented ID.
 - If the row is newly loaded into the table, StarRocks generates a new auto-incremented ID.
 
-This feature can be used to build a dictionary table for accelerating the counting of distinct STRING values.
+This feature can be used to build a dictionary table for fastly computing distinct count of STRING values.
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/auto_increment.md:259]: image
这里动词是 compute 不是 accelerate 了，所以宾语应该是 values 而不是 counting。改为 -> fastly computing distinct STRING values

File: docs/sql-reference/sql-statements/auto_increment.md [hunk 6]
```diff
@@ -320,11 +323,15 @@ This feature can be used to build a dictionary table for accelerating the counti
 
 ## Limits
 
-- When a table with an `AUTO_INCREMENT` column is created, `"replicated_storage" = "true"` must be set to ensure that all replicas have the same auto-incremented IDs.
+- When a table with an `AUTO_INCREMENT` column is created, `'replicated_storage' = 'true'` must be set to ensure that all replicas have the same auto-incremented IDs.
 - Each table can have only one `AUTO_INCREMENT` column.
 - The data type of the `AUTO_INCREMENT` column must be BIGINT.
 - The `AUTO_INCREMENT` column must be `NOT NULL` and does not have a default value.
-- If the `AUTO_INCREMENT` column is the primary key or part of the primary key, you can delete data in the Primary Key table. Otherwise, you cannot delete data in the Primary Key table.
+- You can delete data from the Primary Key table with an `AUTO_INCREMENT` column. However, if the `AUTO_INCREMENT` column is not the primary key or part of the primary key, you need to note the following limits when you delete data:
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/auto_increment.md:330]: image
非特指，改为 an

**Review by amber-create — DISMISSED** [review]

File: docs/sql-reference/sql-statements/auto_increment.md [hunk 7]
```diff
@@ -59,34 +59,37 @@ SELECT * FROM t;
...
+|    2 | 100001 |
 |    3 |    100 |
 +------+--------+
 3 rows in set (0.01 sec)
 ```
 
 Moreover, explicitly specifying values does not affect the subsequent values generated by StarRocks for newly inserted data rows.
+> **NOTICE**
+>
+> It is recommended that you do not use explicit assignment and implicit specification together. Because the specified values may be the same as the values generated by StarRocks, breaking the [global uniqueness of auto-incremented IDs](#uniqueness).
```
Comment (evelynzhaojie) [inline docs/sql-reference/sql-statements/auto_increment.md:79]: 准确性问题：中文是隐式分配和显式指定，英文写反了

File: docs/sql-reference/sql-statements/auto_increment.md [hunk 8]
```diff
@@ -320,11 +323,15 @@ This feature can be used to build a dictionary table for accelerating the counti
...
 ## Limits
 
-- When a table with an `AUTO_INCREMENT` column is created, `"replicated_storage" = "true"` must be set to ensure that all replicas have the same auto-incremented IDs.
+- When a table with an `AUTO_INCREMENT` column is created, `'replicated_storage' = 'true'` must be set to ensure that all replicas have the same auto-incremented IDs.
 - Each table can have only one `AUTO_INCREMENT` column.
 - The data type of the `AUTO_INCREMENT` column must be BIGINT.
 - The `AUTO_INCREMENT` column must be `NOT NULL` and does not have a default value.
-- If the `AUTO_INCREMENT` column is the primary key or part of the primary key, you can delete data in the Primary Key table. Otherwise, you cannot delete data in the Primary Key table.
+- You can delete data from a Primary Key table with an `AUTO_INCREMENT` column. However, if the `AUTO_INCREMENT` column is not the primary key or part of the primary key, you need to note the following limits when you delete data:
+
```
Comment (evelynzhaojie) [inline docs/sql-reference/sql-statements/auto_increment.md:331]: when you delete data in the following scenarios:

File: docs/sql-reference/sql-statements/auto_increment.md [hunk 9]
```diff
@@ -320,11 +323,15 @@ This feature can be used to build a dictionary table for accelerating the counti
...
+  - During the DELETE operation, there is also a load job for partial updates, which only contains UPSERT operations. If both the UPSERT and DELETE operations hit the same data row and the UPSERT operation is executed after the DELETE operation, the UPSERT operation may not take effect.
+  - There is a load job for partial updates, which includes multiple UPSERT and DELETE operations on the same data row. If a certain UPSERT operation is executed after the DELETE operation, the UPSERT operation may not take effect.
+
 - Adding the `AUTO_INCREMENT` attribute by using ALTER TABLE is not supported.
 - Currently, StarRocks's shared-data mode does not support the `AUTO_INCREMENT` attribute.
-- Setting the starting value and increment of the `AUTO_INCREMENT` column is not supported.
+- Do not support specifying the starting value and step size for the `AUTO_INCREMENT` column.
```
Comment (evelynzhaojie) [inline docs/sql-reference/sql-statements/auto_increment.md:337]: 这样写是个祈使句，语气不太好，建议 StarRocks does not support xxx.

File: docs/sql-reference/sql-statements/auto_increment.md (hunk 7 above)
Comment (hellolilyliuyi) [reply docs/sql-reference/sql-statements/auto_increment.md:79]: 这两个是并列的 不算是准确性问题哈 不过为了中英文统一 我改下哈

**Review by evelynzhaojie — APPROVED** [review]

**Review by amber-create — APPROVED** [review]

**Comment by hellolilyliuyi** [comment]
@mergify backport branch-3.0

**Pull request merged**